From 617aca45f31553a2d4f037f0f134b29fd87ead51 Mon Sep 17 00:00:00 2001
From: Narendhar Reddy <quic_nare@quicinc.com>
Date: Fri, 15 Dec 2023 12:57:47 +0530
Subject: [PATCH] wifi: ath12k: Update DP irq names during registration for
 affinity

Update the dp irq names for QCN6432 as there is a conflict in
irq names in case of RDP479 and RDP481.
In ath12k driver irq names for PCI is in the "pci%u_xxxx_%d" where
%u is based on the bus_id for PCI. For QCN6432 %u is based on 
userpd_id.
In case of RDP479 and RDP481 we have one QCN9274 and one QCN6432,
which is leading to conflict as both the bus_id and userpd_id
is 1.Changed the irq_name of QCN6432 from "pci%u_xxx" 
to "pcic%u_xxx".

Signed-off-by: Narendhar Reddy <quic_nare@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/pcic.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/pcic.c b/drivers/net/wireless/ath/ath12k/pcic.c
index a7391e3..a8b3eee 100644
--- a/drivers/net/wireless/ath/ath12k/pcic.c
+++ b/drivers/net/wireless/ath/ath12k/pcic.c
@@ -78,6 +78,7 @@ static const char *irq_name[ATH12K_IRQ_NUM_MAX] = {
 };
 
 char dp_irq_name[ATH12K_MAX_PCI_DOMAINS + 1][ATH12K_EXT_IRQ_DP_NUM_VECTORS][DP_IRQ_NAME_LEN] = {};
+char dp_pcic_irq_name[ATH12K_MAX_PCI_DOMAINS + 1][ATH12K_EXT_IRQ_DP_NUM_VECTORS][DP_IRQ_NAME_LEN] = {};
 char ce_irq_name[ATH12K_MAX_PCI_DOMAINS + 1][ATH12K_IRQ_NUM_MAX][DP_IRQ_NAME_LEN] = {};
 
 void ath12k_pcic_config_static_window(struct ath12k_base *ab)
@@ -419,12 +420,12 @@ static int ath12k_pcic_ext_config_gic_msi_irq(struct ath12k_base *ab,
 			netif_napi_add_weight(&irq_grp->napi_ndev, &irq_grp->napi,
 		       ath12k_pcic_ext_grp_napi_poll, budget);
 
-			scnprintf(dp_irq_name[userpd_id][i], DP_IRQ_NAME_LEN,
-				  "pci%u_wlan_dp_%u", userpd_id, i);
+			scnprintf(dp_pcic_irq_name[userpd_id][i], DP_IRQ_NAME_LEN,
+				  "pcic%u_wlan_dp_%u", userpd_id, i);
 			irq_set_status_flags(msi_desc->irq, IRQ_DISABLE_UNLAZY);
 			ret = devm_request_irq(&pdev->dev, msi_desc->irq,
 					       ath12k_pcic_ext_interrupt_handler, IRQF_SHARED,
-					       dp_irq_name[userpd_id][i], irq_grp);
+					       dp_pcic_irq_name[userpd_id][i], irq_grp);
 			if (ret) {
 				ath12k_err(ab, "failed request irq %d: %d\n", irq_idx, ret);
 				return ret;
-- 
2.34.1

